WordPress/openverse
# Introduce and use new API domain environment variables

## Fixes



Related to #3821 by @sarayourfriend 

## Description


This PR includes the following changes, all resulting from cleaning up hard-coded usages of the API domains:

- Add the two new `CANONICAL_DOMAIN` and `ALTERNATIVE_DOMAINS` variables. Derive an additional setting `CANONICAL_ORIGIN`, which includes the protocol.
- Remove `BASE_URL` and update `ALLOWED_HOSTS` to use the new environment variables and no longer read from `ALLOWED_HOSTS` environment variable
- Update all hard-coded references of `api.openverse.engineering` to `settings.CANONICAL_ORIGIN` or `CANONICAL_DOMAIN` as appropriate for the context in which it is used
- Remove example API client ID, secret, and tokens from request documentation, in favour of using a placeholder
  - This isn't strictly necessary for this PR, but I did a bunch of clean up to the examples anyway. I recall this coming up during our most recent documentation review early in 2023 ([milestone here](https://github.com/WordPress/openverse/milestone/6)), but I can't find an issue for it. In any case, I think it's misleading and potentially suggests something incorrect about these values, that they aren't something users should keep secure. Having placeholders that are clearly such, rather than "stand in" values that _look_ real, makes it clear to users that they should replace their own values here, rather than copy/pasting them to use for themselves (which shouldn't work anyway).
- Remove code supporting the integration test against the "live" API. I don't think this is something that we ever actually use (@dhruvkb you made the most recent changes to these tests, are you aware of us running them against the live API, by chance?) and the configuration relied on hard-coded values for API domains for which there was no easy way to untangle them. The `show_env_name` function was theoretically used, but when I run `just api/test`, the log line it's meant to print never actually shows up, so I don't think it was doing anything anyway. If we want to run these tests against live environments, we can evaluate a different way of doing so (and in all honesty, I think there are probably better approaches, like a k6 suite we actually invest time into).

A knock on effect of all of this, is that the documentation examples _finally_ reflect the environment domain rather than always using the production domain! If you check out https://api-staging.openverse.engineering, or run `main` and read the API docs locally, you'll see that it always references the production domain. This was hard coded, and now no longer is. In all environments, it will now refer to the "canonical" domain for that environment.

Merging this PR is blocked on https://github.com/WordPress/openverse-infrastructure/pull/805, which needs to be deployed before we can merge this PR.

## Testing Instructions

CI should pass, including all tests. Run the application locally and check out the docs to confirm they look good.

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
